src/ray/common/asio/instrumented_io_context.h

```diff
@@ -61,6 +61,23 @@ struct GuardedGlobalStats {
   mutable absl::Mutex mutex;
 };
 
+/// An opaque stats handle, used to manually instrument event loop handlers that don't
+/// hook into a .post() call.
+struct StatsHandle {
+  std::string handler_name;
+  int64_t start_time;
+  std::shared_ptr<GuardedHandlerStats> handler_stats;
+  std::shared_ptr<GuardedGlobalStats> global_stats;
+
+  StatsHandle(std::string handler_name_, int64_t start_time_,
```

Comment:
```suggestion
  StatsHandle(std::string &handler_name_, int64_t start_time_,
```

Reply:
I think that this should be by copy in the constructor argument and `std::move`d when storing it in the struct, similar to the shared pointer comment I made below this.